controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java

```diff
@@ -143,6 +144,20 @@ public void initializeStreamWriters(final EventStreamClientFactory clientFactory
         writerInitFuture.complete(null);
     }
 
+    /**
+     * Method to create stream with retries for lock failed exception.
+     *
+     * @param scope           scope.
+     * @param stream          stream name.
+     * @param config          stream configuration.
+     * @param createTimestamp creation timestamp.
+     * @return CompletableFuture which when completed will have creation status for the stream.
+     */
+    public CompletableFuture<CreateStreamStatus.Status> createStreamWithReries(String scope, String stream, StreamConfiguration config, long createTimestamp) {
```

Comment:
I don't understand why we need a new call `createStreamWithReries`. Can't we make the `createStream` call retryable directly? Why do you want to have one retryable and one non-retryable?

Reply:
CreateStream has annotations which is used by task framework. LockException is thrown by task framework which relies on those annotations as the method with annotations is called. So the retry block has to be on the caller of createstream method. hence the wrapper method.